src/cli.rs

```diff
@@ -1,102 +1,86 @@
-                    Arg::with_name("output_dir")
-                        .short("o")
-                        .long("output-dir")
-                        .takes_value(true)
-                        .help("Outputs assets of the generated site in the given path (by default 'public' dir in project root). HTML/XML will be stored in memory."),
-                    Arg::with_name("base_url")
-                        .short("u")
-                        .long("base-url")
-                        .takes_value(true)
-                        .help("Changes the base_url (default: 127.0.0.1)"),
-                    Arg::with_name("drafts")
-                        .long("drafts")
-                        .takes_value(false)
-                        .help("Include drafts when loading the site"),
-                    Arg::with_name("open")
-                        .short("O")
-                        .long("open")
-                        .takes_value(false)
-                        .help("Open site in the default browser"),
-                    Arg::with_name("fast")
-                        .short("f")
-                        .long("fast")
-                        .takes_value(false)
-                        .help("Only rebuild the minimum on change - useful when working on a specific page/section"),
-                ]),
-            SubCommand::with_name("check")
-                .about("Try building the project without rendering it. Checks links")
-                .args(&[
-                    Arg::with_name("drafts")
-                        .long("drafts")
-                        .takes_value(false)
-                        .help("Include drafts when loading the site"),
-                ])
-        ])
+use std::path::PathBuf;
+
+use clap::{Parser, Subcommand};
+
+#[derive(Parser)]
+#[clap(version, author, about)]
+pub struct Cli {
+    /// Directory to use as root of project
+    #[clap(short, long, default_value = ".")]
+    pub root: PathBuf,
+
+    /// Path to a config file other than config.toml in the root of project
+    #[clap(short, long)]
+    pub config: Option<PathBuf>,
+
+    #[clap(subcommand)]
+    pub command: Command,
+}
+
+#[derive(Subcommand)]
+pub enum Command {
+    /// Create a new Zola project
+    Init {
+        /// Name of the project. Will create a new directory with that name in the current directory
+        #[clap(default_value = ".")]
+        name: String,
+
+        /// Force creation of project even if directory is non-empty
+        #[clap(short, long)]
+        force: bool,
+    },
+
+    /// Deletes the output directory if there is one and builds the site
+    Build {
+        /// Force the base URL to be that value (defaults to the one in config.toml)
+        #[clap(short = 'u', long)]
+        base_url: Option<String>,
+
+        /// Outputs the generated site in the given path (by default 'public' dir in project root)
+        #[clap(short, long)]
```

Comment:
Can the `short` always be named? I don't really want to rely on struct field names because those could be changed and it would only break silently, while the flags are the public api from the cli pov.

Reply:
Sure, sounds reasonable.